jurion.js

```diff
@@ -0,0 +1,383 @@
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+
+function detectWeb(doc, url) {
+	url = decodeURI(url);
+	if (url.indexOf('/?user_nvurlapi_pi1[did]=')>-1) {
+		var type = ZU.xpathText(doc, '//div[contains(@class, "persist-area")]/div[contains(@class, "doc_header")]/@class');
+		if (type.indexOf('artikel_header')>-1) {
+			return "journalArticle";
+		}
+		if (type.indexOf('aktuelles_header')>-1) {
+			return "newspaperArticle";
+		}
+		if (type.indexOf('gesetz_header')>-1) {
+			return "statute";
+		}
+		if (type.indexOf('urteil_header')>-1) {
+			return "case";
+		}
+		if (type.indexOf('stw_header')>-1) {//lexika
+			return "encyclopediaArticle";
+		}
+		if (type.indexOf('hb_header')>-1) {
+			return "bookSection";
+		}
+		if (type.indexOf('jk_header')>-1) {//kommentar
+			return "bookSection";
+		}
+		Z.debug(type);
+	} else if (url.indexOf('/suche/?query=')>-1 && getSearchResults(doc, true)) {
+		return "multiple";
+	}
+}
+
+
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = ZU.xpath(doc, '//ul[contains(@class, "nvurlapi-result-list")]//h2/strong/a');
+	for (var i=0; i<rows.length; i++) {
+		var href = rows[i].href;
+		var title = ZU.trimInternal(rows[i].textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (!items) {
+				return true;
+			}
+			var articles = [];
+			for (var i in items) {
+				articles.push(i);
+			}
+			ZU.processDocuments(articles, scrape);
+		});
+	} else {
+		scrape(doc, url);
+	}
+}
+
+
+function scrape(doc, url) {
+	var type = detectWeb(doc, url);
+	var item = new Zotero.Item(type);
+	
+	item.title = ZU.xpathText(doc, '//div[@id="middle-column"]//h1[div[contains(@class, "addToList")]]');// [contains(@class, "artikel_titel") or contains(@class, "gesetz_ev_titel")]
+	var subtitle = ZU.xpathText(doc, '//div[contains(@class, "artikel_untertitel_thema")]');
+	if (subtitle) {
+		item.title += ' ' + subtitle;
+	}
+	
+	var author = getValueToLabel(doc, "Autor") || getValueToLabel(doc, "Verfasst von");
+	if (author) {
+		author = author.replace(/Dr\.?|Prof\.?|RA\.?/g, '');
+		item.creators.push(ZU.cleanAuthor(author, "author"));
+	}
+	
+	var journal = getValueToLabel(doc, "Zeitschrift");
+	if (journal) {
+		var pos = journal.indexOf('-');
+		if (pos>-1) {
+			item.publicationTitle = journal.substring(pos+1);
+			item.journalAbbreviation = journal.substring(0, pos);
+		} else {
+			item.publicationTitle = journal;
+		}
+	}
+	
+	var ref = getValueToLabel(doc, "Referenz");
+	if (ref) {
+		var m = ref.match(/(\d\d\d\d),\s+([\d\s\-IVX]*)\((.*)\)/);
```

Comment:
Any reason you use `\d\d\d\d` over `\d{4}`? I find the latter much easier to read, but maybe that's just personal preference -- I won't insist.